src/tests/ftest/util/mdtest_test_base.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import os
+
+from apricot import TestWithServers
+from daos_api import DaosPool
+from mpio_utils import MpioUtils
+from mdtest_utils import MdtestCommand, MdtestFailed
+
+class MdtestBase(TestWithServers):
```

Comment:
A good amount of this code is verify similar to ior_test_base.py.  We probably should pull out the common parts (`init` (where `mdtest_cmd` and `ior_cmd` could just be `test_cmd`), `tearDown`, `get_job_manager_cmd`, `create_pool` (see #674), and `run_*_with_pool`) into a more generic command based base test class and add a to-be-overridden `run_test_cmd(self, manager, process)` method.